src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkyWindowRefService
+} from '@skyux/core';
+
+import {
+  SkySummaryActionBarType
+} from './types';
+
+@Injectable()
+export class SkySummaryActionBarAdapterService implements OnDestroy {
+
+  private renderer: Renderer2;
+
+  private resizeSubscription: Subscription;
+
+  constructor(
+    private rendererFactory: RendererFactory2,
+    private windowRef: SkyWindowRefService
+  ) {
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
+
+  public ngOnDestroy() {
+    if (this.resizeSubscription) {
+      this.resizeSubscription.unsubscribe();
+    }
+  }
+
+  public adjustWindowMarginForActionBar(destroying?: boolean): void {
+    const window = this.windowRef.getWindow();
+    const body = window.document.body;
+    if (destroying) {
+      this.renderer.setStyle(body, 'margin-bottom', '');
+    } else {
+      const actionBarEl = <HTMLElement>window.document.getElementsByClassName('sky-summary-action-bar').item(0);
+      this.renderer.setStyle(body, 'margin-bottom', actionBarEl.offsetHeight + 'px');
+    }
+  }
+
+  public setupResizeListener(): void {
+    const windowObj = this.windowRef.getWindow();
+    this.resizeSubscription = Observable
+      .fromEvent(windowObj, 'resize')
+      .subscribe(() => {
+        this.adjustWindowMarginForActionBar();
+      });
+  }
+
+  public removeResizeListener(): void {
+    this.resizeSubscription.unsubscribe();
+  }
+
+  public getSummaryActionBarType(el: Element): SkySummaryActionBarType {
+    do {
```

Comment:
Is there not an easier way to determine the summary action bar's type? Seems like this property should be easily accessible from a component, or a view/content child...

Reply:
I have made some fairly large changes to the main component file to make the type visible to consumers and remove some extra public information that didn't need to be out in the wild.